packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -80,11 +91,31 @@ export interface DatePickerProps
    * e.g. disabledDaysOfWeek={[DayOfWeek.Mon, DayOfWeek.Tue]}
    */
   disabledDaysOfWeek?: DayOfWeek[]
+  /**
+   * Callback when a date is attempted to be selected.
+   * ValidationResponse object will be returned.
+   * @returns {ValidationResponse}
+   */
+  onValidate: (validationResponse: ValidationResponse) => void
+  /**
+   * Updates the styling of the validation FieldMessage.
+   */
+  status: FieldMessageStatus
+  /**
+   * A descriptive message for the 'status' states.
+   */
+  validationMessage: string | React.ReactNode
```

Comment:
```suggestion
  status: FieldMessageStatus | undefined
  /**
   * A descriptive message for the 'status' states.
   */
  validationMessage: string | React.ReactNode | undefined
```

Since we're making them required fields (for now), we need to define that is it valid for them to be `undefined`.